lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2544,6 +2659,25 @@ namespace {
 
         auto VD = cast<VarDecl>(member);
 
+        if (isa<clang::IndirectFieldDecl>(nd) || decl->isUnion()) {
+          // Don't import unconditionnaly unavailable fields that have
+          // no associated storage.
+          bool isUnconditionallyUnavailable = false;
+          auto attrs = VD->getAttrs().getAttributes<AvailableAttr, false>();
+
+          for (auto attr : attrs) {
+            if (attr->isUnconditionallyUnavailable()) {
+              isUnconditionallyUnavailable = true;
+              break;
+            }
+          }
+          if (isUnconditionallyUnavailable) {
+            continue;
+          }
+        }
```

Comment:
This is not strictly equivalent since using `isUnavailable()` will cause the fields to be hidden for any unavailability reason (not just unconditional unavailability).